### PR TITLE
#630 : filerefs now take parent group location into account

### LINKED ISSUE #630: workspace.load_schemes() not finding schemes in a project that is in a group
If I have a workspace with a project that is inside a group, the schemes in that project are not found with workspace.load_schemes()

This is causing problems in fastlane, which cannot access these schemes.

This zip has a workspace with 2 projects, one in a group, and one at root.  

[WorkspaceWIthProjectInGroup.zip](https://github.com/CocoaPods/Xcodeproj/files/2650810/WorkspaceWIthProjectInGroup.zip)

I have the following script (using xcodeproj 1.7.0)

```
require 'xcodeproj'
project_path = './WorkspaceWIthProjectInGroup.xcworkspace/'
workspace = Xcodeproj::Workspace.new_from_xcworkspace(project_path)
workspace.load_schemes(project_path)
print workspace.schemes.to_s.gsub(",", "\n")
```

This prints
```
{"DummyRootProject-WORKS"=>"/Users/lfranco/src/WorkspaceWIthProjectInGroup/DummyRootProject/DummyRootProject.xcodeproj"
 "DummyRootProject"=>"/Users/lfranco/src/WorkspaceWIthProjectInGroup/WorkspaceWIthProjectInGroup.xcworkspace/DummyRootProject/DummyRootProject.xcodeproj"
 "DummyProject"=>"/Users/lfranco/src/WorkspaceWIthProjectInGroup/WorkspaceWIthProjectInGroup.xcworkspace/DummyProject/DummyProject.xcodeproj"}
```

But DummyProject has another shared scheme (DummyProject-NEW).  Here is the result of a find for xcscheme files

```
$ find . -name '*.xcscheme'
./DummyProject/DummyProject/DummyProject.xcodeproj/xcshareddata/xcschemes/DummyProject-NEW.xcscheme
./DummyProject/DummyProject/DummyProject.xcodeproj/xcshareddata/xcschemes/DummyProject.xcscheme
./DummyRootProject/DummyRootProject.xcodeproj/xcshareddata/xcschemes/DummyRootProject-WORKS.xcscheme
./DummyRootProject/DummyRootProject.xcodeproj/xcshareddata/xcschemes/DummyRootProject.xcscheme
```

And here is xcodebuild -list

```
$ xcodebuild -list
Information about workspace "WorkspaceWithProjectInGroup":
    Schemes:
        DummyProject
        DummyProject-NEW
        DummyRootProject
        DummyRootProject-WORKS
```

It really looks to me like `FileReference.absolute_path` is wrong for `type=group` -- it doesn't take the group location into account.  Here is the relevant Group/FileRef node from the workspace

```
   <Group
      location = "container:DummyProject"
      name = "DummyProject">
      <FileRef
         location = "group:DummyProject/DummyProject.xcodeproj">
      </FileRef>
   </Group>
```

The path to the project is `./DummyProject/DummyProject/DummyProject.xcodeproj`

This is called from `Workspace.load_schemes`

### PR DESCRIPTION
**Pull request by kingfai**
This should fix #630.
CC @loufranco

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* #630 : filerefs now take parent group location into account  
  [kingfai](https://github.com/kingfai)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/xcodeproj/workspace/group_reference.rb [hunk 1]
```diff
@@ -17,25 +17,29 @@ class GroupReference
       #
       attr_reader :type
 
+      # @return [String] the location of the group on disk
+      attr_reader :location
```
Comment (dnkoutso) [inline lib/xcodeproj/workspace/group_reference.rb]: nit add `#` line

**Comment by dnkoutso** [comment]
Please also add `CHANGELOG` entry

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,7 +18,9 @@
 
 ##### Bug Fixes
 
-* None.  
+* filerefs now take parent group location into account
```
Comment (segiddins) [inline CHANGELOG.md]: `Workspace file references ...`

File: lib/xcodeproj/workspace/group_reference.rb [hunk 3]
```diff
@@ -46,16 +51,18 @@ def hash
       # @return [GroupReference] The new group reference instance.
       #
       def self.from_node(xml_node)
-        type = xml_node.attribute('location').value.split(':', 2).first
+        location_array = xml_node.attribute('location').value.split(':', 2)
+        type = location_array.first
+        location = (location_array.length > 1) ? location_array[1] : ''
```
Comment (segiddins) [inline lib/xcodeproj/workspace/group_reference.rb]: `location = location_array[1] || ''`

File: lib/xcodeproj/workspace/group_reference.rb [hunk 4]
```diff
@@ -17,25 +17,30 @@ class GroupReference
       #
       attr_reader :type
 
+      # @return [String] the location of the group on disk
+      #
+      attr_reader :location
+
       # @param [#to_s] name @see name
       # @param [#to_s] type @see type
```
Comment (segiddins) [inline lib/xcodeproj/workspace/group_reference.rb]: document the `location` parameter?

**Comment by kingfai** [comment]
Hi all,

Just wondering what the next steps are for this PR? Thanks!

**Pull request merged**